ckan/config/middleware/common_middleware.py

```diff
@@ -28,6 +29,30 @@ def __call__(self, environ, start_response):
         return self.app(environ, start_response)
 
 
+class CloseWSGIInputMiddleware(object):
+    '''
+    webob.request.Request has habit to create FakeCGIBody. This leads(
+    during file upload) to creating temporary files that are not closed.
+    For long lived processes this means that for each upload you will
+    spend the same amount of temporary space as size of uploaded
+    file additionally, until server restart(this will automatically
+    close temporary files).
+
+    This middleware is supposed to close such files after each request.
+    '''
+    def __init__(self, app, config):
+        self.app = app
+
+    def __call__(self, environ, start_response):
+        wsgi_input = environ['wsgi.input']
+        if isinstance(wsgi_input, FakeCGIBody):
+            upload = wsgi_input.vars.get('upload')
```

Comment:
These file objects could be passed as any parameter, not just 'upload'. Can you make this fix more general?

Reply:
Good point. For example the manage-organization image/icon upload is called "image_upload".